src/client/clientmedia.cpp

```diff
@@ -239,10 +252,10 @@ void ClientMediaDownloader::initialStep(Client *client)
 		// requests, so if there are lots of remote servers that are
 		// not responding, those will stall new media file transfers.
 
-		for (u32 i = 0; i < m_remotes.size(); ++i) {
-			assert(m_httpfetch_next_id == i);
+		assert(m_httpfetch_next_id == 0);
 
-			RemoteServerStatus *remote = m_remotes[i];
+		for (u32 i = 0; i < m_remotes.size(); ++i) {
+			auto *remote = m_remotes[i];
```

Comment:
In this case, `auto` is no longer self-explaining. I'd prefer to keep `RemoteServerStatus`. The struct name is unlikely to change anytime soon. Static analysis (e.g. web view) is also more pleasant with explicit types.